Messaging.md

```diff
@@ -359,35 +362,104 @@ Note that `outcome_i` is the outcome value itself and not its hash that will be
    * [`int32`:`precision`]
    * [`u16`:`nb_digits`]
 
+### The `oracle_event_timestamp` Type
+
+This type contains timestamp information about when an `oracle_event` is to happen.
+
+There are two kinds of `oracle_event_timestamps`: fixed timestamps for events with a
+known (expected) epoch time at which the event will occur, and range for events whose
+time is not known ahead of time and also for events which may occur at any time in the
+specified range.
+
+When using an `oracle_event_timestamp_range` for events which may occur at any time,
+the `earliest_expected_time_epoch` should correspond to the time at which the oracle
+announcement was created and `latest_expected_time_epoch` should correspond to the
+latest time for which the oracle will be responsible for attesting to the event's occurence.
+Note that it is possible to have functionally open-ended events through the use of very large
+`latest_expected_time_epoch` timestamps.
+
+#### Fixed `oracle_event_timestamp`
+
+1. type: 55348 (`oracle_event_timestamp_fixed`)
+2. data:
+   * [`u32`:`expected_time_epoch`]
+
+#### Range `oracle_event_timestamp`
+
+1. type: 55350 (`oracle_event_timestamp_range`)
+2. data:
+   * [`u32`:`earliest_expected_time_epoch`]
+   * [`u32`:`latest_expected_time_epoch`]
+
 ### The `oracle_event` Type
 
 This type contains information provided by an oracle on an event that it will attest to.
 See [the Oracle specifications](./Oracle.md#oracle-event) for more details.
 
-#### Version 0 `oracle_event`
+#### Version 1 `oracle_event`
+
+1. type: 55352 (`oracle_event`)
 
-1. type: 55330 (`oracle_event_v0`)
 2. data:
+
    * [`u16`:`nb_nonces`]
    * [`nb_nonces*x_point`:`oracle_nonces`]
-   * [`u32`:`event_maturity_epoch`]
+
+   * [`oracle_event_timestamp`:`timestamp`]
    * [`event_descriptor`:`event_descriptor`]
    * [`string`:`event_id`]
 
+### The `oracle_keys` Type
+
+This type contains static oracle information and can be used to import trusted oracles into wallets.
+This information should be saved by DLC nodes.
+
+#### Version 0 `oracle_keys`
+
+1. type: 55356 (`oracle_keys`)
+2. data:
+   * [`x_point`:`announcement_public_key`]
+   * [`x_point`:`attestation_public_key`]
+   * [`string`:`oracle_name`]
+   * [`string`:`oracle_description`]
+   * [`u32`:`timestamp`]
+   * [`signature`:`announcement_pok_signature`]
+   * [`signature`:`attestation_pok_signature`]
+
+where both proof-of-knowledge signatures are Schnorr signatures of over a sha256 hash of
+`announcement_public_key || attestation_public_key || oracle_name || oracle_description || timestamp`
```

Comment:
It'd be better to just lift the announcement_public_key and signature out of this structure and sign the standard serialization of the inner structure. It is in general bad practice to naively concatenate things when some of the things are variable length (strings) because you don't have a marker for when one starts and the other ends. The serialization naturally gives this to you so it's easier to just sign that.